app/components/request/config-form.js

```diff
@@ -0,0 +1,69 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { task } from 'ember-concurrency';
+import { reads } from '@ember/object/computed';
+import BranchSearching from 'travis/mixins/branch-searching';
+import { bindKeyboardShortcuts, unbindKeyboardShortcuts } from 'ember-keyboard-shortcuts';
+
+const CONFIG = {
+  SOURCE: {
+    API: 'api'
+  },
+  JAVASCRIPT: 'javascript',
+  YAML: 'yaml',
+  JSON: 'json',
+};
+
+export default Component.extend(BranchSearching, {
+  classNames: ['request-configs-form'],
+
+  repo: reads('request.repo'),
+
+  keyboardShortcuts: {
+    'shift+enter': 'submit'
+  },
+
+  onChange() {},
+  onSubmit() {},
+
+  didInsertElement() {
+    this._super(...arguments);
+    bindKeyboardShortcuts(this);
+  },
+
+  willDestroyElement() {
+    this._super(...arguments);
+    unbindKeyboardShortcuts(this);
+  },
+
+  searchBranches: task(function* (query) {
+    const result = yield this.searchBranch.perform(this.get('repo.id'), query);
```

Comment:
It shouldn't be necessary to use `get` here.

```suggestion
    const result = yield this.searchBranch.perform(this.repo.id), query);
```

Reply:
i don't know why, but if i remove `get` here then the branch selector wouldn't make any api request at all, and never find any branches. if i keep it then it works ...

Reply:
looking at this again i have found this ember warning in the log:

```
config-form.js:53 Error: Assertion Failed: You attempted to access the `id` property (of <(unknown):ember843>).
Since Ember 3.1, this is usually fine as you no longer need to use `.get()`
to access computed properties. However, in this case, the object in question
is a special kind of Ember object (a proxy). Therefore, it is still necessary
to use `.get('id')` in this case.
```

Reply:
i suppose `request` is a proxy, and therefore `repo` is?